client/src/components/payment/receipts-modal.vue

```diff
@@ -33,39 +33,48 @@ import PaymentSessionMixin from '@/components/payment/payment-session-mixin'
 @Component({
   components: {
     PaymentSummary
-  },
-  computed: {
-    isVisible: () => PaymentModule[PaymentTypes.PAYMENT_HISTORY_MODAL_IS_VISIBLE]
   }
 })
 export default class ReceiptsModal extends Mixins(NameRequestMixin, PaymentMixin, PaymentSessionMixin) {
   /** Used to display a fetch error, if any. */
   protected fetchError = ''
 
-  @Watch('isVisible')
-  onModalShow (val: boolean, oldVal: string): void {
-    this.fetchData()
-      .then(() => {
-      })
+  /** The model value for the dialog component. */
+  private isVisible = false
+
+  /** Whether this modal should be shown (per store property). */
+  private get showModal (): boolean {
+    return PaymentModule[PaymentTypes.PAYMENT_HISTORY_MODAL_IS_VISIBLE]
   }
 
-  async showModal () {
-    await PaymentModule.togglePaymentHistoryModal(true)
+  /** Depending on value, fetches data and makes this modal visible or hides it. */
+  @Watch('showModal')
+  private async onShowModal (val: boolean) {
+    if (val) {
+      // only make visible on success, otherwise hide it
+      if (await this.fetchData()) {
+        this.isVisible = true
+      } else {
+        await this.hideModal()
+      }
+    } else {
+      this.isVisible = false
+    }
   }
```

Comment:
I rewrote this so that the store property no longer directly sets this dialog visible or not. Now the store property tells this code what to do. This allows us to hide this dialog if there is a fetch error.